### ISSUE TITLE
Cryptic error from gcloud kms command

### ISSUE THREAD

**Issue by charith-qubit** [GoogleCloudPlatform/cloud-builders#144]
I have a build configuration that decrypts a secret using KMS as the first step. Everything works perfectly when I run the build through `container-builder-local` but when run inside GCB, I get the following error message:

```
ERROR: (gcloud.kms.decrypt) INVALID_ARGUMENT: Decryption failed
```

Container builder service account has the "Cloud KMS CryptoKey Decrypter" role attached to it. One thing I noticed when trying to debug the issue is that the version of the gcloud container pulled by container builder is not the same as the one I have locally:

**Container builder**: `gcr.io/cloud-builders/gcloud@sha256:7db6edb77a55f63f2881ef21bf08eb7335f57b5f30cc1d4a83b7cace69cf4aa1`
**Local**: `gcr.io/cloud-builders/gcloud@sha256:8f4c285c2ced181762a36619a0625926289aac7fbce4edc4d2653e6678fba3e7`

I updated my build config to use the container version that works locally. However, the build still fails with the same error. 

```yaml
steps:
    - name: gcr.io/cloud-builders/gcloud@sha256:8f4c285c2ced181762a36619a0625926289aac7fbce4edc4d2653e6678fba3e7
      args:
          - --project=$PROJECT_ID
          - kms
          - decrypt
          - --ciphertext-file=/workspace/_build.tar.gz.enc
          - --plaintext-file=/credentials/_build.tar.gz
          - --location=global
          - --keyring=xxx-keyring
          - --key=xxx-key
```

I am a bit lost as it's impossible to recreate the problem locally and the remote error message is not very descriptive. Any thoughts on what could be causing this?

**Comment by ImJasonH** [comment]
`gcloud@sha256:7db6edb` runs gcloud 174.0.0, while `gcloud@sha256:8f4c285` runs gcloud 175.0.0. So there's something different in these two releases of gcloud that's causing this. [gcloud release notes](https://cloud.google.com/sdk/docs/release-notes#17500_2017-10-11) don't indicate anything noteworthy.

How large is the archive you're decrypting? KMS has a 64KiB limit for encrypt/decrypt operations, but I would have expected this to be enforced in either release.

I'll ping someone from the KMS team to see if this is a known issue.

**Comment by ImJasonH** [comment]
Also, you might get more information by adding `--verbosity=debug` and `--log-http` flags to your gcloud invocations. **Please** make sure you scrub any logging output of sensitive data and auth tokens if you post it here.

**Comment by charith-qubit** [comment]
Thank you! That helped me zero in on the issue. It was a PEBKAC. Turns out I had created the same key ring in a testing project and the file was encrypted with the key from that project instead of the project on which GCB was running. Since my local gcloud was defaulting to the testing project, everything worked fine when running `container-builder-local`. That coupled with the "INVALID ARGUMENT" error from KMS really threw me off the scent.
